ios/browser/api/history/brave_history_api.mm

```diff
@@ -0,0 +1,309 @@
+}
+@end
+
+#pragma mark - BraveHistoryAPI
+
+@interface BraveHistoryAPI () <BraveHistoryDriverDelegate> {
+  // Browser State Used to retreive History Driver - History Service and Web
+  // History Service
+  ChromeBrowserState* browser_state_;
+  // History Browsing Service Driver
+  std::unique_ptr<BraveBrowsingHistoryDriverIOS> browsing_history_driver_;
+  // History Browsing Service
+  std::unique_ptr<history::BrowsingHistoryService> browsing_history_service_;
+  // History Service for adding and querying
+  history::HistoryService* history_service_;
+  // WebhistoryService for remove and elete operations
+  history::WebHistoryService* web_history_service_;
+  // Tracker for history requests.
+  base::CancelableTaskTracker tracker_;
+}
+@property(nonatomic, strong) void (^query_completion)(NSArray<IOSHistoryNode*>*)
+    ;
+@end
+
+@implementation BraveHistoryAPI
+
+/// Shared Singleton Instance for History API
++ (instancetype)sharedHistoryAPI {
+  static BraveHistoryAPI* instance = nil;
+  static dispatch_once_t onceToken;
+
+  dispatch_once(&onceToken, ^{
+    instance = [[BraveHistoryAPI alloc] init];
+  });
+
+  return instance;
+}
+
+/// Constructor which is setting up services
+/// like HistoryService / SyncService / BrowsingHistoryService /
+/// WebHistoryService
+- (instancetype)init {
+  if ((self = [super init])) {
+    DCHECK_CURRENTLY_ON(web::WebThread::UI);
+    ios::ChromeBrowserStateManager* browserStateManager =
+        GetApplicationContext()->GetChromeBrowserStateManager();
+    browser_state_ = browserStateManager->GetLastUsedBrowserState();
+    browsing_history_driver_ =
+        std::make_unique<BraveBrowsingHistoryDriverIOS>(browser_state_, self);
+    history_service_ = ios::HistoryServiceFactory::GetForBrowserState(
+        browser_state_, ServiceAccessType::EXPLICIT_ACCESS);
+    syncer::SyncService* sync_service =
+        ProfileSyncServiceFactory::GetForBrowserState(browser_state_);
+    browsing_history_service_ =
+        std::make_unique<history::BrowsingHistoryService>(
+            browsing_history_driver_.get(), history_service_, sync_service);
+
+    web_history_service_ =
+        ios::WebHistoryServiceFactory::GetForBrowserState(browser_state_);
+    DCHECK(history_service_);
+    DCHECK(browsing_history_service_);
+  }
+  return self;
+}
+
+- (void)dealloc {
+  browser_state_ = nil;
+  history_service_ = nil;
+  web_history_service_ = nil;
+}
+
+- (id<HistoryServiceListener>)addObserver:(id<HistoryServiceObserver>)observer {
+  return [[HistoryServiceListenerImpl alloc] init:observer
+                                   historyService:history_service_];
+}
+
+- (void)removeObserver:(id<HistoryServiceListener>)observer {
+  [observer destroy];
+}
+
+- (bool)isLoaded {
+  // Triggers backend to load if it hasn't already, and then returns whether
+  // it's finished loading.
+  return history_service_->BackendLoaded();
+}
+
+- (void)addHistory:(IOSHistoryNode*)history {
```

Comment:
this seems like it could easily be misused. I think when importing existing data we should explicitly set it to TYPED

Reply:
Method is removed. So the page transition should be set every-time by using the other `addHistory` method on iOS side.